src/lazy/iterable.ts

```diff
@@ -0,0 +1,65 @@
+import _ from "lodash";
+import { traverseWith, AnyNode } from "../traverse";
+import { ObjectKey, SeedIterable } from "./abstract";
+import hasType from "./hasType";
+
+const isIterable = <T>(maybe: unknown): maybe is Iterable<T> => typeof (maybe as Iterable<T>)[Symbol.iterator] === "function";
+const getValues = <T>(seed: SeedIterable<T, never>): Iterable<T> => {
+	if (hasType.set<T>(seed)) return seed;
+	if (hasType.lazySet<T>(seed)) return seed;
+	if (hasType.lazyArray<T>(seed)) return seed;
+	if (hasType.array(seed)) return seed;
+	if (hasType.lazyObject(seed)) return seed.iterators.values();
+	if (seed instanceof Map) return seed.values();
+	if (isIterable<T>(seed)) return seed;
```

Comment:
can a switch statement with two clauses being `hasType.lazyObject(seed)` and `seed instanceof Map` be used here to return their specific properties and then having a default clase where `seed` will be returned?

Or is it that this is setup in this way to provide more clarity?

Reply:
IDK, I went with this https://github.com/jakehamtexas/csnow/pull/1/commits/931f531b145cafa725bc29a56a2261145d564566